### PR TITLE
Add Gridfinity Refined thumbscrew option

### PR DESCRIPTION
**Pull request by andrewguy**
I've implemented the thumbscrew option from Gridfinity Refined (https://www.printables.com/model/413761-gridfinity-refined), enabling an optional screw hole in the middle of each base unit.

I did this for my own benefit, but I noticed there was an open issue for this (#195).

I'm very new to OpenSCAD (only been a couple of days), so I may not have the best approach here, but it seems to work.

I've pulled in Ryan Colyer's threads-scad library to handle this - https://github.com/rcolyer/threads-scad. Included as a top-level `threads.scad` file. Not sure if this is how you want to handle library dependencies?

These thumbscrew holes print well on a Bambu Lab A1 mini with Bambu PLA Basic, and the thumbscrew file provided by Gridfinity Refined work well (engages well, easy to insert and minimal slack).

gridfinity_thumbscrew

### CODE REVIEW

**Comment by EmperorArthur** [comment]
We need to discuss how we want to handle libraries.

There are four major options I can think of:
* Require the user to manually have it.  -- Obviously bad idea.
* Include it in something like an "External" folder.  -- Library updates involve copying files.
* Use git submodules. -- Need to add extra instructions for people pulling down code.
* Do not allow 3rd party libraries.

If 3rd party libraries are allowed, we will need to make sure they are compatible license wise.

So far I have deliberately stayed away from libraries, but if we do decide to use them, then I suggest we use [BSOL](https://github.com/revarbat/BOSL) or [BSOL2](https://github.com/BelfrySCAD/BOSL2/).  Either would replace both this threading library, and a significant amount of helper code.


# Changes Needed
Regardless, there will need to be changes to this PR.  Even if it is just putting the library in a separate folder, and making a note of where to get the latest version.

File: threads.scad
Comment (EmperorArthur) [inline threads.scad:1]: See my latest comment.  At the least this needs to be in it's own folder, and the Readme updated with where this was downloaded from.

File: gridfinity-rebuilt-utility.scad
Comment (EmperorArthur) [inline gridfinity-rebuilt-utility.scad:298]: If you'll take a look at what the `ScrewHole` module does, you'll see it's a wrapper around `diference(){children() ScrewThread(...)}` with some translations that aren't used here.

Since this is already inside a `difference()` statement, there's no need for a wrapper, and it both duplicates code and doesn't fit with how holes are handled in this function.

I'd suggest putting the if statement and a raw `ScrewThread(...)` call immediately before or after the section with the comment "4 holes".

File: gridfinity-rebuilt-utility.scad
Comment (EmperorArthur) [inline gridfinity-rebuilt-utility.scad:274]: Don't forget to add the additional parameter as a comment, and an assertion to ensure the value being passed in is a bool.

File: gridfinity-rebuilt-utility.scad
Comment (EmperorArthur) [inline gridfinity-rebuilt-utility.scad:217]: Good job here.

File: gridfinity-rebuilt-utility.scad
Comment (EmperorArthur) [inline gridfinity-rebuilt-utility.scad:206]: Good job here.

**Review by EmperorArthur — CHANGES_REQUESTED** [review]
I like the idea, but there are a few tweaks I'd like to see.

The big thing is we need to determine how libraries are handled.

File: gridfinity-rebuilt-utility.scad
Comment (andrewguy) [reply gridfinity-rebuilt-utility.scad]: Thanks, I've refactored this to just do a call to ScrewHole() as suggested. Works well.

**Comment by andrewguy** [comment]
Thanks @EmperorArthur for the feedback.

I've refactored into a call to ScrewHole(), although this introduces a few more 'magic numbers' that come from the `threads.scad` code (i.e. `1.01*outer_diam + 1.25*tolerance`, presuming this introduces a bit of slack for threading the screw). Worth it I think.

I've moved the `threads.scad` library into `/external/threads-scad/`. In terms of licence, `threads.scad` is released under a [CC0-1.0 license](https://github.com/rcolyer/threads-scad#CC0-1.0-1-ov-file), which means it should be fine to include here.

In terms of handling external libraries, the only other option which would allow this pull request to go through would be to use Git submodules. I'd be concerned that submodules would lead to more trouble than they're worth (e.g. people need to use `--recurse-submodule` if doing a `git clone`, which often isn't obvious).

If the decision is made to not use external libraries, obviously this PR is a non-starter, which is fine.

File: gridfinity-rebuilt-utility.scad
Comment (EmperorArthur) [inline gridfinity-rebuilt-utility.scad:299]: I'd prefer these in standard.scad, but will move them myself.

**Review by EmperorArthur — APPROVED** [review]
Personally I'm more of a fan of git submodules, but for something like this that doesn't change often I don't see a problem with it.  I also brought up the license as a future thing, not something to worry about right now.

Approving this, since my fix to #225 will have conflicts with this code, and I don't want to trash your hard work.

**Comment by EmperorArthur** [comment]
@andrewguy Since I'm not a maintainer I can't edit this pull request, and I'm not allowed to manually merge it.

Could you please fix the whites-pace issue so the checks pass.  I'll merge this once that's done.  You could also squash your commits, or I can have the merge do it.

**Comment by andrewguy** [comment]
Whitespace is fixed and all the pre-commit checks pass. If you could squash on merge that would be great - that's my normal strategy, and I don't want to break something by trying to squash an existing branch. :sweat_smile: 

Thanks for the review and being happy to integrate this. Appreciate it.

**Pull request merged**
